samza-core/src/main/scala/org/apache/samza/storage/ContainerStorageManager.java

```diff
@@ -239,66 +253,102 @@ public ContainerStorageManager(
     this.taskStores = createTaskStores(containerModel, jobContext, containerContext, storageEngineFactories, serdes, taskInstanceMetrics, taskInstanceCollectors);
 
     // create system consumers (1 per store system in changelogSystemStreams), and index it by storeName
-    Map<String, SystemConsumer> storeSystemConsumers = createConsumers(this.changelogSystemStreams.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey,
-        e -> Collections.singleton(e.getValue()))), systemFactories, config, this.samzaContainerMetrics.registry());
+    Map<String, SystemConsumer> storeSystemConsumers = createConsumers(this.changelogSystemStreams.values().stream().map(SystemStream::getSystem).collect(
+        Collectors.toSet()),
+        systemFactories, config, this.samzaContainerMetrics.registry());
+
     this.storeConsumers = createStoreIndexedMap(this.changelogSystemStreams, storeSystemConsumers);
 
     // creating task restore managers
     this.taskRestoreManagers = createTaskRestoreManagers(systemAdmins, clock, this.samzaContainerMetrics);
 
     // create sideInput storage managers
-    sideInputStorageManagers = createSideInputStorageManagers(clock);
+    this.taskSideInputStorageManagers = createSideInputStorageManagers(clock);
+
+    this.taskSideInputHandlers = new HashMap<>();
 
-    // create sideInput consumers indexed by systemName
-    this.sideInputConsumers = createConsumers(this.sideInputSystemStreams, systemFactories, config, this.samzaContainerMetrics.registry());
+    Set<String> sideInputSystems = this.taskSideInputStoreSSPs.values().stream()
+        .flatMap(map -> map.values().stream())
+        .flatMap(Set::stream)
+        .map(SystemStreamPartition::getSystem)
+        .collect(Collectors.toSet());
+
+    this.sideInputSSPLocks = this.taskSideInputStoreSSPs.values().stream()
+        .flatMap(map -> map.values().stream())
+        .flatMap(Collection::stream)
```

Comment:
can we extract this to local variable and reuse this above for fetching `sideInputSystems` too? looks like its just doing the same.

Reply:
Good point. Yes, we can extract this to a variable.